### DISCUSSION TITLE
How $channel->wait() work on loop forever

### DISCUSSION THREAD

**Discussion by Louis-Ni** [php-amqplib/php-amqplib#940] [Q&A]
I am using this library to consume RabbitMQ messages on the Laravel framework, after I wanna go deeper about this library then I confuse on **$channel->wait()** method.

how can I trace who calls this method when the rabbitMQ server sent message to consumer application?

I wrote the code below on the `wait()` method
```
var_dump(debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS), 10);
```
Then I found the result is where I called `wait()` for and no other result

then I traced it deeper I found the message is coming from 
```
protected function dispatch_deferred_method($queued_method)
{
     $this->debug->debug_method_signature('Executing queued method: %s', $queued_method[0]);
     return $this->dispatch($queued_method[0], $queued_method[1], $queued_method[2]);
}
```
but I don't know who trigger `$channel->wait()` does it trigger by system callback?

**Comment by ramunasd** [comment]
Actually it does not. You must add it to your own loop.
Here is example:
https://github.com/php-amqplib/php-amqplib/blob/master/demo/amqp_consumer.php#L83
